client/src/test/scala/drt/client/components/BigSummaryBoxTests.scala

```diff
@@ -353,28 +353,27 @@ object ApiFlightGenerator {
                  iataFlightCode: String = "BA123",
                  Origin: String = "",
                  PcpTime: Long = 0): Arrival =
-    Arrival(
-      Operator = Operator,
+    new Arrival(
+      Operator = Some(Operator),
       Status = Status,
-      EstDT = EstDT,
-      ActDT = ActDT,
-      EstChoxDT = EstChoxDT,
-      ActChoxDT = ActChoxDT,
-      Gate = Gate,
-      Stand = Stand,
-      MaxPax = MaxPax,
-      ActPax = ActPax,
-      TranPax = TranPax,
-      RunwayID = RunwayID,
-      BaggageReclaimId = BaggageReclaimId,
-      FlightID = FlightID,
+      Estimated = if (EstDT != "") Some(SDate(EstDT).millisSinceEpoch) else None,
+      Actual = if (ActDT != "") Some(SDate(ActDT).millisSinceEpoch) else None,
+      EstimatedChox = if (EstChoxDT != "") Some(SDate(EstChoxDT).millisSinceEpoch) else None,
+      ActualChox = if (ActChoxDT != "") Some(SDate(ActChoxDT).millisSinceEpoch) else None,
+      Gate = if (Gate!= "") Some(Gate) else None,
```

Comment:
could you update the signature of this builder to take options rather than updating them in the body.